File: merge_2.cpp
#include <iostream>

using namespace std;

int sorted[8];
const int length = 8;

void merge(int *A, int m, int middle, int n)
{
	int i, j, k;
	i = m;
	j = middle + 1;
	k = m;
	
	while(i <= middle && j <= n)
	{
		if(A[i] <= A[j])
		{
			sorted[k] = A[i];
			i++;
		} else 	{
			sorted[k] = A[j];
			j++;	
		}
		k++;
	}
	
	if(i > middle)
	{
		for(int t = j; t <= n; t++)
		{
			sorted[k] = A[t];
			k++;
		}
	} else {
		for(int t = i; t <= middle; t++)
		{
			sorted[k] = A[t];
			k++;
		}
	}
	
	for(int t = 0; t < n; t++)
	{
		A[t] = sorted[t];
	}
}

void mergeSort(int *A, int m, int n)
{
	
	if(m < n)
	{
		int middle = (m + n) / 2;
		mergeSort(A, m, middle);
		mergeSort(A, middle + 1, n);
		merge(A, m, middle, n);
	}
	
}



int main()
{
	int A[length] = {1,4,2,5,4,7,4,8};
	mergeSort(A,0,7);
	
	for(int i = 0; i < length; i++)
	{
		cout << sorted[i] < " ";
	}
}

File: merge2.py
import random
a=[random.randint(0,50) for k in range(25)]

def mergesort(a):
    if len(a) <= 1:
        return a

    mid = len(a)//2

    left = mergesort(a[:mid])
    right = mergesort(a[mid:])
    i, j = 0, 0
    sorted = []
    while i < len(left) and j < len(right):
        if left[i] < right[j]:
            sorted.append(left[i])
            i += 1
        else:
            sorted.append(right[j])
            j += 1
    
    while j < len(right):
        sorted.append(right[j])
        j += 1
    while i < len(left):
        sorted.append(left[i])
        i += 1
    return sorted

print(mergesort(a))
